src/docfx/build/Build.cs

```diff
@@ -73,12 +76,12 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Erro
         private static async Task Run(
             Docset docset,
             Docset fallbackDocset,
-            RestoreGitMap restoreGitMap,
             CommandLineOptions options,
             ErrorLog errorLog,
-            string outputPath)
+            string outputPath,
+            GitLock gitLock)
         {
-            using (var context = new Context(outputPath, errorLog, docset, fallbackDocset, restoreGitMap))
+            using (var context = new Context(outputPath, errorLog, docset, fallbackDocset, gitLock))
```

Comment:
Changing from RestoreMap to GitLock is a leak of abstraction. GitLock is an internal data structure that should only be known by the restore component. Other parts of the system query restore formation through the `RestoreMap` interface.

Reply:
good point